packages/driver-interface/src/services/bot.ts

```diff
@@ -5,10 +5,24 @@ import { v4 as uuid } from 'uuid'
 import * as messaging from './messaging'
 import { IncomingMessage, Message } from 'telegraf/typings/telegram-types'
 import { TelegrafContext } from 'telegraf/typings/context'
+import { Instruction } from '../types'
 
-export const driverIsLoggedIn = async (vehicleId): Promise<boolean> => {
-  const vehicle = await cache.getVehicle(vehicleId)
-  return !!vehicle.telegramId
+export const driverIsLoggedIn = async (
+  telegramId: number
+): Promise<boolean> => {
+  const vehicleId = await cache.getVehicleIdByTelegramId(telegramId)
+  return !!vehicleId
+}
+
+export const onInstructionsReceived = async (
+  telegramId: number,
+  instructions: Instruction[][]
```

Comment:
Suggestion: In most places, Instruction[][] is named instructionGroups. Goes for many places :)